Initial version of R1/Beta2 release notes. (#376)
haiku/website: content/get-haiku/release-notes-r1b2.md

Please keep in mind that this is beta-quality software, which means it is feature complete but still contains known and unknown bugs. While we are mostly confident in its stability, we cannot provide assurances against data loss.

*To download Haiku, see "[Get Haiku!](/get-haiku/)". You may also upgrade an existing Beta 1 installation using the [instructions](#upgrading-from-beta-1). For press inquiries, see "[Press contact](#press-contact)".*

## Table of Contents
 - [System requirements](#system-requirements)
 - [Upgrading from Beta 1](#upgrading-from-beta-1)
 - [Beta 2 Highlights](#beta-2-highlights)
 - [Other enhancements](#other-enhancements)
 - [Changes for Developers and Porters](#changes-for-developers-and-porters)
 - [New Contributor](#new-contributor)
 - [Source Code](#source-code)
 - [Reporting Issues](#reporting-issues)
 - [Press contact](#press-contact)

## System requirements
This release is available on the x86 32-bit platform, as well as the x86_64 platform. Note that BeOS R5 compatibility is only provided on the 32-bit images.

<table><tr><td>

> <h4>MINIMUM (32-bit)</h4>

Reviewer: Maybe also include a Maximum section?  :-)  Seriously, maximum 1 IDE controller (4 hard drives), else it won't work.  Had years of disappointment that Haiku wouldn't run on my Pentium III PC, when it was just too many hard drive controllers.

Reviewer 2: That's just a bug (and only affects a very small number of computers) so I don't think it warrants mentioning here.

<ul>

> <li><strong>Processor</strong>: Intel Pentium II; AMD Athlon</li>

Reviewer 3: This is a bit restrictive, the Pentium Pro and the AMD K6 would also work, and so would Via C3.

> <li><strong>Memory:</strong> 256MB</li>

Reviewer 3: 256MB seems barely enough to boot the Desktop (after a long time) and it is not possible to start even AboutSystem apparently.

<li><strong>Monitor:</strong> 800x600</li>

> <li><strong>Storage:</strong> 3GB</li>

Reviewer 3: The minimal install (without enabling any optional packages) needs a lot less than that.

</ul>
</td><td>
<h4>RECOMMENDED (64-bit)</h4>
<ul>
<li><strong>Processor</strong>: Intel Core i3; AMD Phenom II</li>

> <li><strong>Memory:</strong> 2GB</li>

Reviewer 4: 512MB should be fine as a minimum I think?

Reviewer 2: This is the "recommended" section.

<li><strong>Monitor:</strong> 1366x768</li>

> <li><strong>Storage:</strong> 16GB</li>

Reviewer 3: 4 or 8GB would probably be pleinty enough?

</ul>
</td></tr></table>

SSE2 support is required to use the WebPositive web browser. On machines where this is not available, it is recommended to install the NetSurf browser instead.

Haiku takes advantage of modern hardware and will run faster on newer machines. 

## Upgrading from Beta 1

> There is an upgrade path available for users that are currently using Haiku R1 Beta 1. If you want to do this upgrade, you will have to use the Terminal to issue some commmands.

Reviewer 3: for users who are ?

*Warning: only upgrading from Beta 1 is supported. It may be possible to upgrade from other intermediate development images, but this is untested and you may run into unknown problems.*

### Step 1: Getting the Latest Updates

> You need to make sure that you have all the package updates that are available for your current release. You can do this by running the following command in the terminal:

Reviewer 3: It is not possible to do this as the beta1 repos are currently offline, if I understand correctly.

```shell script

> pkgman update

Reviewer 4: I think `pkgman full-sync` would be more safe.

Reviewer 2: I don't think this section is actually needed at all.

```

### Step 2: Switch to the new repositories
After this we will switch the package manager to point to the new repositories for beta 2. This will allow the package manager to find the new packages. Run the following commands in the Terminal.

```shell script
pkgman drop Haiku; pkgman drop HaikuPorts;
pkgman add-repo https://eu.hpkg.haiku-os.org/haiku/r1beta2/$(getarch)/current
pkgman add-repo https://eu.hpkg.haiku-os.org/haikuports/r1beta2/$(getarch)/current
```

### Step 3: Installing the new packages and reboot

Now it is time to download and install the packages. You can do this by issuing the following command in the Terminal:

```shell script
pkgman full-sync
```

After that step has been successful, you should immediately reboot the system. You can do this from the Deskbar, or by running ```shutdown -r``` in the terminal.

## Beta 2 Highlights

This release contains 20 months worth of improvements. Below you find some highlights.

> ### New Input Preferences

Reviewer 4: I think it's better to start with something more exciting than a preflet :)

Reviewer 2: Agreed.

The Mouse, Keyboard, Touchpad and Joystick preferences have been consolidated into one Input Preferences.

<img src="/files/get-haiku/input_preferences.png"/><br>
<small><i>The consolidated Input Preferences, including support for mice with 5 buttons.</i></small>

Additionally, there is now support for mice with more than three buttons.

> ### DeskBar improvements

Reviewer 4: We should either stick to Deskbar or DeskBar. I think the former is more regular.

Reviewer 2: Agreed.

> The Deskbar has had various improvements. It now has a mini-mode, where the icons on the shelf expand leftwards. You can activate it by dragging the shelf space by the right handle to the top of the screen.

Reviewer 3: It already had a mini mode, the new one just looks different (it puts everything on one line, before there would be one line with deskbar menu + teams menu, and the shelf below)

<img src="/files/get-haiku/deskbar_minimode.png"/><br>
<small><i>New mini-mode to put the Deskbar at the top of the screen.</i></small>

> Furthermore, there are more options for showing larger icons, which is especially useful on screens with a higher DPI.

Reviewer 4: Wasn't it in beta1 already?

Reviewer 5: I think so, too

<img src="/files/get-haiku/deskbar_icon_sizes.png"/><br>
<small><i>More size option for icons in DeskBar.</i></small>

### Hardware Support: xHCI, hda, intel_extreme and NVMe

Haiku's contributors have improved on xHCI, the host controller for USB3. This should make USB3 devices more stable. It also fixes some issues with booting Haiku from a USB3 storage device.

Likewise, the Intel High Definition Audio (`hda`) driver has received some improvements, increasing the number of chipsets that it supports.

> The driver for Intel graphics chipsets (`intel_extreme`) has had various improvements. It supports more chipsets, as well as blacklists chipsets that are known not to work. The advantage to the latter is that a user will no longer have to manually disable the driver in the boot menu when their system uses one of the blacklisted chipsets.

Reviewer 3: It's not really a blacklist, the unsupported chips are just removed from the driver list of known hardware. But I don't have a better wording, so...

> Finally, initial support for NVMe storage devices have been implemented. This is a protocol for modern SSD drives, which takes advantage of the properties of this modern storage medium, to improve I/O performance over previous storage protocols. Reminder: Haiku is still in beta quality, and there may be bugs in these storage drivers that cold cause data loss. 

Reviewer 4: s/cold/could

### Installation Improvements

There are various improvements in the installation process. The first is that there is better support for EFI, though work remains to be done. Additionally, the Installer now supports optional packages that are included on the Installation medium. 

DriveSetup will now display more information about the drives in your system. It will show the used space of existing partitions, and it is now also able to identify encrypted volumes.

Finally, there is now also an upgrade path available from Haiku R1 beta 1 to beta 2.

<img src="/files/get-haiku/installer_optionalpackages.png"/><br>
<small><i>Optional package selection in the Installer.</i></small>

### The Terminal now has an emulated Meta-key

> Certain early workstation keyboards included a key labeled "Meta" that functioned as an extra modifier key alongside Shift and Control. Its purpose was to give users an easy way of entering special characters: When held down it caused the eighth bit to be set on all characters read from the keyboard, allowing access to ​the "extended" portion of the ASCII character set (character codes above 127). 

Reviewer 3: I would say this bit of UNIX history seems a bit irrelevant in the release notes. People who don't know what a Meta key is will not gain much from its support in Terminal, anyway

Reviewer 4: Same though here.

Although the Meta key is absent from modern keyboards, two major pieces of UNIX software still rely on it being present: GNU Emacs and the GNU readline library, which is used by bash and a wide variety of other software to read input from the terminal. (Python's interactive shell uses readline, for instance.) These applications use the Meta key as a "command" key and provide additional editing features when it's available. 

In beta 2, the Terminal will have functionality like macOS': It adds a configuration option to the "Settings..." dialog that, when enabled, causes the left Option key (only) to function as a Meta key. The right Option key retains its normal function, and can be used to enter special characters at the keyboard. 

<img src="/files/get-haiku/terminal_metakey.png"/><br>
<small><i>Enabling the meta key functionality in the settings.</i></small>

## Other Enhancements

### Applications
**Deskbar**

>  - [#4971](https://dev.haiku-os.org/ticket/4971) Resize Deskbar from the Deskbar settings, our by using the CTRL-ALT-right click drag

Reviewer 4: This was implemented a long time ago, even before beta 1.

 - [#8537](https://dev.haiku-os.org/ticket/8537) In menus, the size of the icon will now always be the same size as your chosen font size
 - [#8691](https://dev.haiku-os.org/ticket/8691) The icons of replicants are running in the Deskbar, will scale according to the settings
 - [#13304](https://dev.haiku-os.org/ticket/13304) Auto-raised deskbar will now also auto-unraise

**DriveSetup**

>  - [#10098](https://dev.haiku-os.org/ticket/10098) Encrypted disks will actually display that they are encrypted instead of empty

Reviewer 3: this is already mentionned in the "improved installation process" section

> **HaikuDepot**

Reviewer 3: Doesn't all the work on HaikuDepot deserve a section in the "highlights"? To me it seems much more important than the meta key, for example. Well, I guess having an highlight section can only lead to bikeshedding about what goes there.

 - [#10358](https://dev.haiku-os.org/ticket/10358) When a package is queued for download and installation, it will now have the status 'queued' in the interface.
 - [#11652](https://dev.haiku-os.org/ticket/11652) When showing all packages, the installed packages are part of the list by default. Before they were hidden by default. 

>  - [#12198](https://dev.haiku-os.org/ticket/12198) HaikuDepot can now display multiple screenshots, when the package has multiple to show.

Reviewer 4: Again, this has already worked in beta 1.

 - [#12428](https://dev.haiku-os.org/ticket/12428) It is possible to switch between a view with all tabs, and a view with featured packages, by using the tabs at the top of the window.
 - [#13806](https://dev.haiku-os.org/ticket/13806) The application will now clearly show when it is downloading screenshots for the packages you are looking at.
 - [#13808](https://dev.haiku-os.org/ticket/13808) Overall, it is more clear when HaikuDepot is contacting the web to retrieve information or updates.

**Installer**
 - [#14310](https://dev.haiku-os.org/ticket/14310) The installer now supports the user showing and selecting optional packages, when they are available on the installation medium

**Launchbox**
 - [#11237](https://dev.haiku-os.org/ticket/11237) Launchbox will now start automatically at boot, when that has been configured in its settings.
 - [#13835](https://dev.haiku-os.org/ticket/13835) Icons of 96x96 and 128x128 pixels are now supported.

**MediaPlayer**

>  - [#6628](https://dev.haiku-os.org/ticket/6628) When MediaPlayer is opened by opening a media file from Tracker, there will be a mark at the timestamp at the point where the user last stopped listening/viewing that file. 

Reviewer 4: Not sure, probably was in beta 1 too.

**People**
 - [#9397](https://dev.haiku-os.org/ticket/9397) It is possible to quickly create a new Person file from Tracker, with the 'Create new...' feature

> **SoftwareUpdate**

Reviewer 3: SoftwareUpdater (mising r)

>  - [#14002](https://dev.haiku-os.org/ticket/14002) When updating the system packages itself, SoftwareUpdater will now display a 'Restart' button to inform the user that a restart is necessary.

Reviewer 3: Is that the only change? If SoftwareUpdater is available in beta1, why does the update instruction go through Terminal? If it's not available, the changelog here should just mention that it is a new addition to beta2. (also check availability of the Repositories preferences for a fully GUI based update)

**Terminal**
 - [#15294](https://dev.haiku-os.org/ticket/15294) Support for an emulated Meta key

**Tracker**
 - [#1264](https://dev.haiku-os.org/ticket/1264) multi-range selection in Tracker and list views
 - [#3011](https://dev.haiku-os.org/ticket/3011) Directories with a large number of files are now loaded quicker
 - [#5126](https://dev.haiku-os.org/ticket/5126) Improvements to the Tracker API

>  - [#10529](https://dev.haiku-os.org/ticket/10529) For files that are part of a package, show information about that package in the info window.

Reviewer 3: The original description of the ticket is quite restrictive: this has grown into a complete rewrite of the "File information" window and it can display all extended attributes, not just package specific information.

**WebPositive**
 - [#14845](https://dev.haiku-os.org/ticket/14845) Allow selecting any font size (was previously limited to 18pt).

**Command Line Utilities**

>  - [#10288](https://dev.haiku-os.org/ticket/10288) `uname` now shows the specific revision of a Haiku nightly build.

Reviewer 4: This was implemented long before beta 1.

>  - [#14845](https://dev.haiku-os.org/ticket/14845) Allow selecting any font size (was previously limited to 18pt).

Reviewer 3: It's not clear from this description, 18pt was the maximum size for the default font in Web+. It was of course possible to use other font sizes: smaller ones for the default size; and smaller and larger ones as expected from website's css.

>  - [#12665](https://dev.haiku-os.org/ticket/12665) the built in `ftp` is now replaced with [https://ftp.netbsd.org/pub/pkgsrc/current/pkgsrc/net/tnftp/README.html tnftp]

Reviewer 6: ```suggestion
 - [#12665](https://dev.haiku-os.org/ticket/12665) The builtiin `ftp` was replaced with [https://ftp.netbsd.org/pub/pkgsrc/current/pkgsrc/net/tnftp/README.html tnftp].
```

### Drivers

>  - The `hda` driver has had some improvements, that will allow it to work properly on more devices. If you had problem in R1/beta1, then it is worth a shot to try it again.

Reviewer 6: ```suggestion
 - The `hda` driver has had some improvements to work properly on more devices. If you had problem in Beta 1, it is worth a shot to try again.
```

 - [#9910](https://dev.haiku-os.org/ticket/9910) NVMe device support

>  - [#5056](https://dev.haiku-os.org/ticket/5056) Intel Extreme: support X4500HD

Reviewer 3: That is very specific. The more important info is that the driver has complete support for all pre-SandyBridge devices, and limited support for SandyBridge and IvyBridge (only LVDS and DDI displays in laptops work for now, VGA, DVI, DisplayPort and HDMI being still on the TODO list).

On older hardware we should handle multiple displays in "clone" mode now (both displays showing the same thing).

 - [#12723](https://dev.haiku-os.org/ticket/12723) Intel Extreme: support hardware panel scaling for chipsets that support it

>  - [#13987](https://dev.haiku-os.org/ticket/13987) FreeBSD Network Drivers: upgrade network drivers to FreeBSD 11.1

Reviewer 3: Mention that this adds compatibility with more ethernet and wifi devices.

### Preferences

>  - [#10171](https://dev.haiku-os.org/ticket/10171) Support configuration of advanced mice and trackballs

Reviewer 3: I think the highlight section about this is more interesting and descriptive and there is no point in linking to a specific ticket here.

 - [View.h](https://git.haiku-os.org/haiku/commit/headers/os/interface/View.h?id=c67dde0f2ccb45d550af410cf6d73fee39d10504) Add Tiled Background methods to `BView` objects to asynchronously draw bitmaps as background tiles for the view, using the overloaded `DrawTiledBitmapAsync()` method.

### Locale Kit
 - [Country.h](https://git.haiku-os.org/haiku/commit/headers/os/locale/Country.h?id=70cdd7d4f5fc62e8b3e220646f84235ec3d444d5) Add a `SetTo()` and an `InitCheck()` method to the `BCountry` class.

### Package Kit
 - [PackageRoster.h](https://git.haiku-os.org/haiku/commit/headers/os/package/PackageRoster.h?id=e2c7bb900cf83fe0a3834ad2189577687ffaa03f) Add a `IsRebootNeeded()` method to the `BPackageRoster` class.
 - [PackageRoster.h](https://git.haiku-os.org/haiku/commit/headers/os/package/PackageRoster.h?id=b711002d345e2e9e0075be46a37492624f73ccdd) Add a `IsPackageActive()` method to the `BPackageRoster` class.

## New contributor

> Since the last release, Kyle Ambroff-Kao received commit access.

Reviewer 3: Does that really belong in the release notes?

Reviewer 2: We mentioned it last time.

## Source code

> The source code of Haiku itself, the source code of the required build tools and the optional packages (except for closed source ones) is contained within the release images as the `_source` packages (except on the "CD" image, which it was left out of due to space constraints.)

Reviewer 3: There is no "CD" image. We release a single DVD/anyboot image that includes the sourcecode.

For the closed source packages I would prefer that we clearly list what this includes, because it's a common FAQ that we're not "completely open source" or something thanks to a mention about that on the FSF website. As far as I know, it's only wifi firmwares, and liblayout which is needed only by Wonderbrush in the release image.

> The source code of Haiku itself, the source code of the required build tools and the optional packages (except for closed source ones) is contained within the release images as the `_source` packages (except on the "CD" image, which it was left out of due to space constraints.)

Reviewer 6: ```suggestion
The source code of Haiku itself, the source code of the required build tools and the optional packages (except for closed source ones) is contained within the release images as the `_source` packages (except on the "CD" image, where it was left out due to space constraints.)
```

> ## Reporting issues

Reviewer 6: ```suggestion
## Reporting Issues
```

There are almost  3200 open tickets on Haiku's bug tracker and over 12700 closed items.  If you find what you believe to be an issue, please search our Trac to see if it has already been reported, and if not, file a new ticket at <https://dev.haiku-os.org/>.

For information about major issues that have been fixed since the release, visit <https://dev.haiku-os.org/wiki/R1/Beta2/ReleaseAddendum>.

> For more help see the 'Welcome' link on the Haiku desktop, or visit the Haiku Project's website at <https://www.haiku-os.org>.

Reviewer 3: There is no link to the welcome page on the desktop anymore. It is however still the start page of webpositive.

> For more help see the 'Welcome' link on the Haiku desktop, or visit the Haiku Project's website at <https://www.haiku-os.org>.

Reviewer 6: 'Welcome' isn't linked on the Desktop anymore.

```suggestion
For more help, take the 'Quick Tour' and read the 'Userguide', both linked on the Haiku desktop. WebPositive opens by default with our 'Welcome' page which provides useful information and many helpful links, as does the Haiku Project's website at <https://www.haiku-os.org>.
```

## Press contact

Press inquiries may be directed to:

 * waddlesplash (English)

>  * pulkomandy (French)

Reviewer 3: PulkoMandy

 * humdingerb (German)

> All three contacts may be reached via `<nickname>@gmail`.

Reviewer 3: My address is pulkomandy@pulkomandy.tk
